app/models/ability.rb

```diff
@@ -0,0 +1,20 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the passed in user here.
+    # See the documentation for details:
+    # https://github.com/CanCanCommunity/cancancan/wiki/Defining-Abilities
+
+    # Any user can create a new Thesis
+    can :create, Thesis
+
+    # Only the Thesis owner can view their Thesis
+    can :read, Thesis, user_id: user.id
```

Comment:
The docs wrap tests of user attributes in an `if user.present?`, but this seems to work even if we don't do that. I guess it's cool? (ISTR rails and django handle anonymous users differently but I don't follow the specifics.)

Reply:
oh. right. Let me dig in on that a bit more as now I'm not sure...hmmm.